students/mattclau/lesson08/circle.py

```diff
@@ -0,0 +1,65 @@
+#Class for creating and manipulating circles
+import math
+from functools import total_ordering
+
+@total_ordering
+class Circle:
+
+    _diameter = 0
+    pi = math.pi
+    _area = 0
+    circles = []
+
+    def __init__(self, radius, diameter=None):
+        self.radius = radius
+        self.diameter = radius*2
+        self.circles.append(self)
```

Comment:
it is not ideal design. Circle represents a single circle and shouldn't have a reference to many circles.

Reply:
to your question about unused diameter param; you're not actually using it in here, you're setting 
`self.diameter = radius * 2`

the alternate constructor should be:
```
    @classmethod
    def from_diameter(cls, diameter):
        return cls(diameter / 2)
```